File: src/main/java/nl/sijpesteijn/lasforce/domain/Animation.java
package nl.sijpesteijn.lasforce.domain;

import org.mongodb.morphia.annotations.Entity;
import org.mongodb.morphia.annotations.Id;
import org.mongodb.morphia.annotations.Property;

/**
 * @author Gijs Sijpesteijn
 */
@Entity("animations")
public class Animation {
    @Id
    @Property("_id")
    private long id;
    private String name;
    private long lastUpdate;
    private int frameRate;
    private int nrOfFrames;
    private String fileName;

    public void setName(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

    public void setLastUpdate(long lastUpdate) {
        this.lastUpdate = lastUpdate;
    }

    public long getLastUpdate() {
        return lastUpdate;
    }

    public int getFrameRate() {
        return frameRate;
    }

    public void setFrameRate(int frameRate) {
        this.frameRate = frameRate;
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getFileName() {
        return fileName;
    }

    public void setFileName(String fileName) {
        this.fileName = fileName;
    }

    public int getNrOfFrames() {
        return nrOfFrames;
    }

    public void setNrOfFrames(int nrOfFrames) {
        this.nrOfFrames = nrOfFrames;
    }
}

File: src/main/java/nl/sijpesteijn/lasforce/domain/Animation.kt
package nl.sijpesteijn.lasforce.domain

import java.util.*

/**
 * @author Gijs Sijpesteijn
 */
data class Segment(var color:List<Int>, var coordinates:List<List<Int>>)
data class Frame(var id:Long, var name: String, var repeat:Int, var total_segments:Int, var segments:List<Segment>)
data class Animation(var id:Long, var name: String, var last_update: Long, var repeat:Int, var total_frames:Int = 0, var frame_time: Int = 0, var frames: List<Frame> = ArrayList<Frame>())
data class AnimationInfo(var id:Long, var repeat:Int)